### PR TITLE
Add method to get users by cohort name

### PR DESCRIPTION
**Pull request by s-espinosa**
@srt32 Hadn't discussed this, but believe this may come in handy for some users as well.

### CODE REVIEW

File: spec/census/client_spec.rb [hunk 1]
```diff
@@ -64,6 +64,63 @@
     end
   end
 
+  describe '#get_users_by_cohort_name' do
+    it 'returns users associated with a cohort' do
```
Comment (srt32) [inline spec/census/client_spec.rb]: Big 🎉  for the specs ;)

File: lib/census/client.rb [hunk 2]
```diff
@@ -89,6 +89,15 @@ def get_users(limit: 100, offset: 0)
       end
     end
 
+    def get_users_by_cohort_name(cohort_name)
+      users_search_url = build_full_url_with_token(path: "/api/v1/users/search_all") + "&1=#{cohort_name}"
```
Comment (srt32) [inline lib/census/client.rb]: Looking at https://github.com/turingschool-projects/census/blob/47917577612677deb34ad63899274fb1194ceb7e/app/controllers/api/v1/users/search_all_controller.rb#L6 I think this param currently written as `1` should really be a `q`. No?

**Comment by srt32** [comment]
This change looks great less the question about the query param name.

Merge whenever you're ready.

A couple questions for us to consider going forwards:

1) I think we'll thank ourselves later if we prefer to search / filter by cohort.id instead cohort.name. Name feels fragile and confusing. We'd have to see what it would take to update the search endpoint to support `id` and also make sure the resources we return to clients have the proper affordances to give the cohort.id so folks can send it back for searches.

2) We currently paginate the users endpoint. Should we paginate this search endpoint? I think if someone sends a really simple filter we could possible return back all the users. It's probably ~1000 records now, but will probably timeout sometime soon.

Thoughts on either of those ideas?

**Comment by srt32** [comment]
And same comment as on your other PR...


Please follow the instructions in https://github.com/turingschool-projects/omniauth-census#releasing-a-new-version once you merge and lemme know if you have any issues!

**Comment by s-espinosa** [comment]
Good catch on the query param name. It makes me wonder if instead of stubbing out the request we should use VCR. Thoughts?

Re: searching by cohort_id - I think it makes sense to have that as an endpoint, but think there is potential benefit to having an endpoint where someone can search by cohort without having to make a query for the cohort id. The app using Census would have to do some error checking to make sure they got a cohort back if a user entered a cohort that didn't exist, but it seems like the alternative is to 1) make a query for all cohorts and then search through them to find the id of the cohort they're looking for, or 2) provide a search by cohort name to get a cohort back and then use the id to search for users (which seems to be pretty much the equivalent of what we're doing, but with an extra step).

Think it makes sense to provide the option to paginate the search endpoint. Going to make a card for that now.

**Comment by srt32** [comment]
If we dint query by id then the calling code has to know the cohort name
somehow. If we are cool assuming cohort names (or the structure of them)
don’t change over time then 👍🏽

**Pull request merged**